hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/TransactionProcessorTest.java

```diff
@@ -370,6 +371,32 @@ void callWhenComputePartiesThrowsException() {
         assertThat(handleException.getStatus()).isEqualTo(INVALID_ACCOUNT_ID);
     }
 
+    @Test
+    void callWhenInvalidContractIdThrowsException() {
+        final var transaction = new HederaEvmTransaction(
+                SENDER_ID,
+                null,
+                INVALID_CONTRACT_ADDRESS,
+                NONCE,
+                CALL_DATA,
+                MAINNET_CHAIN_ID,
+                0L,
+                GAS_LIMIT,
+                USER_OFFERED_GAS_PRICE,
+                MAX_GAS_ALLOWANCE,
+                null,
+                null);
+        final var context = wellKnownContextWith(blocks, tinybarValues, systemContractGasCalculator);
+        given(worldUpdater.getHederaAccount(SENDER_ID)).willReturn(senderAccount);
+        given(worldUpdater.getHederaAccount(INVALID_CONTRACT_ADDRESS)).willThrow(IllegalArgumentException.class);
+
+        final var handleException = catchThrowableOfType(
+                () -> subject.processTransaction(
+                        transaction, worldUpdater, () -> feesOnlyUpdater, context, tracer, config),
+                HandleException.class);
+        assertThat(handleException.getStatus()).isEqualTo(INVALID_TRANSACTION_BODY);
```

Comment:
I thought the point of the code at `ContractCallHandler` L91 is that this should return an `INVALID_CONTRACT_ID` not `INVALID_TRANSACTION_BODY`?  Is someone mapping the `PreCheckException` to something else? and losing the more specific error code?

Reply:
This is for the more general case if an exception is thrown while computing parties in `TransactionProcessor` that is _not_ already a `HandleException`.  In this case, we may not know what threw the exception but that likely the transaction body was malformed